python/dgl/graphbolt/negative_sampler.py

```diff
@@ -38,32 +37,33 @@ def __init__(
         self.negative_ratio = negative_ratio
         self.output_format = output_format
 
-    def _sample(self, node_pairs):
+    def _sample(self, data):
         """
         Generate a mix of positive and negative samples.
 
         Parameters
         ----------
-        node_pairs : Tuple[Tensor] or Dict[etype, Tuple[Tensor]]
-            A tuple of tensors or a dictionary represents source-destination
-            node pairs of positive edges, where positive means the edge must
-            exist in the graph.
+        LinkPredictionBlock : Tuple[Tensor] or Dict[etype, Tuple[Tensor]]
+            An instance of 'LinkPredictionBlock', where 'node_pair' is required
+            while 'negative_head' and 'negative_tail' should be None. If either
+            of them has value, it inidcates there have already negative samples
+            , thus the function will do nothing.
 
         Returns
         -------
         LinkPredictionBlock
             An instance of 'LinkPredictionBlock' encompasses both positive and
             negative samples.
         """
-
-        data = LinkPredictionBlock(node_pair=node_pairs)
-        if isinstance(node_pairs, Mapping):
-            for etype, pos_pairs in node_pairs.items():
-                self._collate(
-                    data, self._sample_with_etype(pos_pairs, etype), etype
-                )
-        else:
-            self._collate(data, self._sample_with_etype(node_pairs))
+        if data.negative_head is None and data.negative_tail is None:
```

Comment:
What if the label exists?

Reply:
I would suggest not checking here, if user call negative_sampler with node_pair, it is ok to override existing data.negative_head and data.negative_tail.

Reply:
Remove check for fix.